app/javascript/book/add_book.js

```diff
@@ -60,4 +45,42 @@ $(function() {
             searchResults.appendChild(bookResultsDiv);
         });
     }
+
+    $('#search_google_book_submit').on('click', function() {
+        autofillBookData(mybook.volumeInfo);
+    });
+
+    function autofillBookData(bookDataObject) {
+        $(".removable_field").remove();
+        $('#book_title').val(bookDataObject.title);
+        $('#publisher_name').val(bookDataObject.publisher);
+        $('#book_description').val(bookDataObject.description);
+        $('#book_isbn').val(bookDataObject.industryIdentifiers[1].identifier);
+        $('#book_page_count').val(bookDataObject.pageCount);
+        $('#book_image_link').val(bookDataObject.imageLinks.thumbnail);
+        $('#book_language').val(bookDataObject.language);
+        bookDataObject.authors.forEach(author => {
+            $authorDiv = $($("#author_field_template").html());
+            $("#authors_field").append($authorDiv);
+            $authorDiv.find('#author_name').val(author);
+        });
+        bookDataObject.categories.forEach(category => {
+            $categoryDiv = $($("#category_field_template").html());
+            $("#categories_field").append($categoryDiv);
+            $categoryDiv.find('#category_name').val(category);
+        });
+    }
+
+    $("#add_author").click(function(){
+        $("#authors_field").append($("#author_field_template").html());
+    });
+
+    $("#add_category").click(function(){
+        $("#categories_field").append($("#category_field_template").html());
+    });
+
+    $('#authors_field, #categories_field')
+    .delegate('input.delete_item', 'click', function() {
```

Comment:
I like using `delegate` jquery function here.

Actually It is the first time I see it. Well done.

Reply:
:)